tests_new/integration_tests/core/environment/global_conftest.py

```diff
@@ -87,5 +86,21 @@ def temp_env1(client1, group1, org1, testdata):
 """
 
 
-@pytest.fixture(scope='function')
-def persistent_env1(client1, group1, org1, testdata): ...  # TODO
+def get_or_create_persistent_env(env_name, client, group, testdata):
+    envs = list_environments(client, term=env_name).nodes
+    if envs:
+        return envs[0]
+    else:
+        envdata = testdata.envs[env_name]
+        org = create_organization(client, f'org_{env_name}', group)
+        env = create_env(client, group, org['organizationUri'], envdata.accountId, envdata.region, tags=[env_name])
+        if env.stack.status == 'CREATE_COMPLETE':
```

Comment:
What about UPDATE_COMPLETE?

Reply:
Every day there is an scheduled task that updates environment and dataset stacks. It is an edge case that exactly at the same time we run our tests, but it could happen. Should we consider it in the design?

Reply:
Although this will be pretty unlikely, polling is 10seconds, but it won't harm adding it. Will also have to add it to the assertions.